File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        unordered_set<int> seen;

        for (int it : nums) {
            if (seen.count(it)) {
                return it;
            }
            seen.insert(it);
        }
        return -1;
    }
};

File: 1001-n-repeated-element-in-size-2n-array/n-repeated-element-in-size-2n-array.cs
public class Solution {
    public int RepeatedNTimes(int[] nums) {
        HashSet<int> seen = new HashSet<int>();

        foreach (int i in nums) {
            if (seen.Contains(i)) {
                return i;
            }
            seen.Add(i);
        }
        
        return -1;
    }
}